app/components/agenda/agendaitem/send-to-vp-modal.js

```diff
@@ -0,0 +1,206 @@
+    );
+    const ONTWERPDECREET = this.documentTypes.find(
+      (type) => type.uri === CONSTANTS.DOCUMENT_TYPES.ONTWERPDECREET
+    );
+    const MEMORIE = this.documentTypes.find(
+      (type) => type.uri === CONSTANTS.DOCUMENT_TYPES.MEMORIE
+    );
+    const NOTA = this.documentTypes.find(
+      (type) => type.uri === CONSTANTS.DOCUMENT_TYPES.NOTA
+    );
+    const ADVIES = this.documentTypes.find(
+      (type) => type.uri === CONSTANTS.DOCUMENT_TYPES.ADVIES
+    );
+
+    const { PRINCIPIELE_GOEDKEURING, DEFINITIEVE_GOEDKEURING } =
+      CONSTANTS.SUBCASE_TYPES;
+
+    this.subcaseRequirements = {
+      [PRINCIPIELE_GOEDKEURING]: [
+        { type: BESLISSINGSFICHE, wordRequired: false, signed: true },
+        { type: ONTWERPDECREET, wordRequired: true, signed: false },
+        { type: MEMORIE, wordRequired: true, signed: false },
+        { type: NOTA, wordRequired: false, signed: false },
+        { type: ADVIES, wordRequired: false, signed: false },
+      ],
+      [DEFINITIEVE_GOEDKEURING]: [
+        { type: BESLISSINGSFICHE, wordRequired: false, signed: true },
+        { type: ONTWERPDECREET, wordRequired: true, signed: true },
+        { type: MEMORIE, wordRequired: true, signed: true },
+        { type: NOTA, wordRequired: false, signed: false },
+        { type: ADVIES, wordRequired: false, signed: false },
+      ],
+    };
+  }
+
+  missingDocsForSubcase = (subcaseWithPieces) => {
+    // We want to move this logic to the backend so it will always be
+    // consistent with the actual data being sent.
+    const formattedMissingFiles = [];
+    const subcaseRequirements =
+      this.subcaseRequirements[subcaseWithPieces.subcaseTypeUri];
+
+    for (const requirement of subcaseRequirements) {
+      if (
+        requirement.signed &&
+        !hasSignedPieceOfType(subcaseWithPieces.pieces, requirement.type)
+      ) {
+        formattedMissingFiles.push(
+          `${requirement.type.altLabel} (ondertekend)`
+        );
+      } else if (
+        !requirement.signed &&
+        !hasPieceOfType(
+          subcaseWithPieces.pieces,
+          requirement.type,
+          'application/pdf; charset=binary'
+        )
+      ) {
+        formattedMissingFiles.push(`${requirement.type.altLabel}`);
+      }
+
+      if (
+        requirement.wordRequired &&
+        !hasPieceOfType(
+          subcaseWithPieces.pieces,
+          requirement.type,
+          // question: is this enough for all Word documents?
+          // or do we need to check more mimeTypes?
+          'application/vnd.openxmlformats-officedocument.wordprocessingml.document; charset=binary'
+        )
+      ) {
+        formattedMissingFiles.push(`${requirement.type.altLabel} (in Word)`);
+      }
+    }
+
+    return formattedMissingFiles;
+  };
+}
+
+function findPieceOfType(pieces, type, mimeType) {
+  return pieces.find((piece) => {
+    const documentContainer = piece.belongsTo('documentContainer').value();
```

Comment:
I'm not a big fan of this way of loading async data. Will do for now, but should be fixed later

Reply:
But this is not loading async data. Could you maybe elaborate on why you're not a big fan?